src/NzbDrone.Common/Disk/DiskTransferService.cs

```diff
@@ -83,6 +84,100 @@ public TransferMode TransferFolder(string sourcePath, string targetPath, Transfe
             return result;
         }
 
+        public int MirrorFolder(string sourcePath, string targetPath)
+        {
+            var filesCopied = 0;
+
+            Ensure.That(sourcePath, () => sourcePath).IsValidPath();
+            Ensure.That(targetPath, () => targetPath).IsValidPath();
+
+            _logger.Debug("Mirror [{0}] > [{1}]", sourcePath, targetPath);
+
+            if (!_diskProvider.FolderExists(targetPath))
+            {
+                _diskProvider.CreateFolder(targetPath);
+            }
+
+            var sourceFolders = _diskProvider.GetDirectoryInfos(sourcePath);
+            var targetFolders = _diskProvider.GetDirectoryInfos(targetPath);
+
+            foreach (var subDir in targetFolders.Where(v => !sourceFolders.Any(d => d.Name == v.Name)))
+            {
+                _diskProvider.DeleteFolder(subDir.FullName, true);
+            }
+
+            foreach (var subDir in sourceFolders)
+            {
+                filesCopied += MirrorFolder(subDir.FullName, Path.Combine(targetPath, subDir.Name));
+            }
+
+            var sourceFiles = _diskProvider.GetFileInfos(sourcePath);
+            var targetFiles = _diskProvider.GetFileInfos(targetPath);
+
+            foreach (var targetFile in targetFiles.Where(v => !sourceFiles.Any(d => d.Name == v.Name)))
```

Comment:
If the filename still exists in the source, regardless of the path it won't be deleted though.

If the target has a file of `\test.txt` and the source has `\test\test.txt` it will won't be deleted and the target directory will have both files.